src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Ecc.cs

```diff
@@ -2,21 +2,63 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
 using System.Runtime.InteropServices;
+using System.Security.Cryptography;
 using System.Security.Cryptography.Apple;
 
 internal static partial class Interop
 {
     internal static partial class AppleCrypto
     {
-        [DllImport(Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_EccGenerateKey")]
-        internal static extern int EccGenerateKey(
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static extern int AppleCryptoNative_EccGenerateKey(
             int keySizeInBits,
+            SafeKeychainHandle tempKeychain,
             out SafeSecKeyRefHandle pPublicKey,
             out SafeSecKeyRefHandle pPrivateKey,
             out int pOSStatus);
 
         [DllImport(Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_EccGetKeySizeInBits")]
         internal static extern long EccGetKeySizeInBits(SafeSecKeyRefHandle publicKey);
+
+        internal static void EccGenerateKey(
+            int keySizeInBits,
+            out SafeSecKeyRefHandle pPublicKey,
+            out SafeSecKeyRefHandle pPrivateKey)
+        {
+            using (SafeTemporaryKeychainHandle tempKeychain = CreateTemporaryKeychain())
+            {
+                SafeSecKeyRefHandle keychainPublic;
+                SafeSecKeyRefHandle keychainPrivate;
+                int osStatus;
+
+                int result = AppleCryptoNative_EccGenerateKey(
+                    keySizeInBits,
+                    tempKeychain,
+                    out keychainPublic,
+                    out keychainPrivate,
+                    out osStatus);
+
+                if (result == 1)
+                {
+                    pPublicKey = keychainPublic;
+                    pPrivateKey = keychainPrivate;
+                    return;
+                }
+
+                using (keychainPrivate)
+                using (keychainPublic)
+                {
+                    if (result == 0)
+                    {
+                        throw CreateExceptionForOSStatus(osStatus);
+                    }
+
+                    Debug.Fail($"Unexpected result from AppleCryptoNative_EccGenerateKey: {result}");
+                    throw new CryptographicException();
```

Comment:
A non-boolean value should mean that we don't trust osStatus to report the true cause of the error (it might even still be `noErr`).